### PR TITLE
Color zones: Make the colors in the interface background less saturated to prevent visual illusions

### LINKED ISSUE #4630: “Color zones” module’s interface has too saturated colors
The “color zones” module is very useful, but the colors in its interface (the saturation tab in particular) are *extremely* saturated, so when I look at them while adjusting the curves and then go back to looking at my image, I have an afterimage of the interface, and the colors in the image look very dull in comparison, leading to excessive usage of the saturation curve if I don’t wait several seconds for my eyes to reacclimate before judging.

Would it be possible to tone down the colors a little bit, please? :)

Thanks.

### PR DESCRIPTION
**Pull request by msdm**
Fixes #4630

### CODE REVIEW

**Comment by TurboGit** [comment]
Can you post screen shot before and after ? I agree the color on this module are very "aggressive" so probably a good enhancement.

**Comment by msdm** [comment]
Sure!
Before:
Screenshot from 2020-12-09 22-26-15
Screenshot from 2020-12-09 22-26-33
Screenshot from 2020-12-09 22-26-48
Screenshot from 2020-12-09 22-27-02
Screenshot from 2020-12-09 22-27-17
Screenshot from 2020-12-09 22-27-29
Screenshot from 2020-12-09 22-27-46
Screenshot from 2020-12-09 22-27-57
Screenshot from 2020-12-09 22-28-04
After:
Screenshot from 2020-12-09 22-30-28
Screenshot from 2020-12-09 22-30-39
Screenshot from 2020-12-09 22-30-49
Screenshot from 2020-12-09 22-31-06
Screenshot from 2020-12-09 22-31-15
Screenshot from 2020-12-09 22-31-24
Screenshot from 2020-12-09 22-31-35
Screenshot from 2020-12-09 22-31-45
Screenshot from 2020-12-09 22-31-56

**Comment by TurboGit** [comment]
Hum... I would say this goes a bit too far. The color are difficult to see now.

**Comment by msdm** [comment]
I have divided the saturation by 8. We can divide it by 4 to make it more colorful. Though I find the latter case yet too colorful particulary when you work with a photo low in saturation.

**Comment by johnny-bit** [comment]
This is good idea, but maybe a bit too much in terms of desaturation... especially when saturation is divided by 8.

Given that we can use events and configuration, how about making it 2 levels: saturation when module is having a focus (eg mouse is inside) and when mouse is outside desaturate module. And have configuration options for saturation divisor when module is in focus and when isn't in forcus.

That way the module wouldn't be "too saturated" most of the time and prestent clear interfcase (with enough saturation) when user interacts with it

Would that be ok?

**Comment by Mark-64** [comment]
I personally prefer saturated colors, as I am slightly color blind and I have problems with subtles hues.
Please do not desaturate too aggressively

**Comment by TurboGit** [comment]
@msdn: given latest comments, can you provide a PR with a saturation divided by 2 only for testing ? TIA.

**Review by TurboGit — CHANGES_REQUESTED** [review]
Please propose a 2x factor of desaturation and also clean-up the patch set you have 48 commits, many of them are from other people !

**Comment by johnny-bit** [comment]
@msdm - please fix your branch... instead of merging master, rebase on top of master... or rather - as long as there aren't any conflicts don't merge/rebase at all.

for now - just rebase your branch on master.

**Comment by msdm** [comment]
@johnny-bit, I messed it up. I'm trying to fix it.

**Comment by msdm** [comment]
@TurboGit, sorry for the mess caused by merging my fork with the upstream. The saturation is divided by 2 now. But I think the bg is still too saturated. Can we make the desaturation/saturation factor somehow configurable (maybe through darktablerc or the css theme)?

**Comment by TurboGit** [comment]
@msdm : can you post new screen shots ?

**Comment by msdm** [comment]
Just updated my previous comment with some screenshots

**Comment by TurboGit** [comment]
To me this is well balanced. We still read easily the colors but I don't feel it is aggressive this way.
I'm not sure we can achieve that with CSS, it would be possible with a darktablerc pref but and hidden one, not in the preferences dialog as we already have too many prefs and I don't feel this is something we want to advertise. But at least you'll be able to tweak that as you like.

**Comment by msdm** [comment]
I completely agree with you about the darktablerc approach without prefs. IMO it's nice to have the ability to fine-tune such detailed configs through darktablerc (similar to about:config of firefox).

**Comment by msdm** [comment]
@TurboGit, I added the darktablerc pref. The default value is 0.5, but I'm not sure I set it in the correct way (is the default value supposed to be set with dt_conf_set_float()?).

File: src/iop/colorzones.c
Comment (TurboGit) [inline src/iop/colorzones.c:988]: This should not be needed. Please add an entry into `data/darktableconfig.xml.in` (no prefs / section attribute).

type should be float, default should be 0.5 and min set to 0.1 and max to 1.

See `plugins/darkroom/masks/heightview` for example.

When this is done you just have to take the value (dt_conf_get_float) and not worry about the initial value and even if 0 since the value will be checked to be in the defined bound each time dt starts.

**Comment by johnny-bit** [comment]
use darktableconfig.xml.in to generate "default" value (it's good to also add descriptions & min&max value) and then using `dt_conf_get_float` will clamp the value between min&max specified in that file! automatic garbage config avoidance :) just make sure that min and max setting allow range from "maximally saturated" to "completelly desaturated" without causing problems :)

**Comment by msdm** [comment]
Thanks, guys. It's fixed now.

File: data/darktableconfig.xml.in
Comment (TurboGit) [inline data/darktableconfig.xml.in:2495]: saturation factor of the color zone background

Comment (msdm) [reply data/darktableconfig.xml.in]: fixed

**Comment by TurboGit** [comment]
Works for me, thanks !

**Comment by TurboGit** [comment]
Merged in master, queued for 3.4.1.

**Comment by todd-prior** [comment]
@TurboGit  @msdm  Instead of guessing all the time or trying to hardwire color etc would there be a way to add a slider to displays like this and the histogram etc that would allow people to brighten them to their taste. Given the range of displays that people use and the way they are set up...both davinci and rawtherapee allow this and its great...also for my eyes I like bright saturated colors when using the control ...others may want it dim...but its nice to just have a small slider to add or subtract some boost.....

**Comment by TurboGit** [comment]
@todd-prior : that's not practical, we don't want a sliders in all place of dt to adjust GUI rendering. That would obscure the UI. Here we have an hidden prefs for those really wanting to change the default. I think it is the best compromise.

**Comment by todd-prior** [comment]
I'm just glad I noticed this so I can look for the pref and alter it....likely good to revisit that file and review all the settings anyway...Thx

**Comment by johnny-bit** [comment]
I think this is prime candidate for @phweyland's work for #7050 in form of PR #7294 :)

Once PR #7294 lands I have a couple of usability ideas...

**Pull request merged**
